### ISSUE TITLE
EVURLCache and Alamofire

### ISSUE THREAD

**Issue by evermeer** [evermeer/EVURLCache#25]
Reported by @nmDejamobile via email.

Hi,

I found your EVURLCache lib and it looks nice as it does all the work for me :)
But I use Alamofire and I can’t figure out how to get the result of the request.
When cache is involed, Alamofire .Success or .Failure callback are not called.
However I well see the log stating that the cached data was loaded by EVURLCache.
Also, after sometime, Alamofire .Failure is triggered by a timeout.

Any ideas on how I could use it ?
Thanks.

**Comment by evermeer** [comment]
Well, it seems to just work when I try it. When I try one of the sample Alamofire callse:

```
        Alamofire.request(.GET, "https://httpbin.org/get", parameters: ["foo": "bar"])
            .validate(statusCode: 200..<300)
            .validate(contentType: ["application/json"])
            .response { response in
                print("\(response)")
        }
```

Then the first time it's called, I will see this in the log:

```
07/28/2016 22:54:43:203 Oauth2Test)[83082:.] EVURLCache.swift(85) cachedResponseForRequest:
	PRECACHE not found /Users/evermeer/Library/Developer/CoreSimulator/Devices/544A81ED-E3EB-40C9-B098-A695DF6B6A81/data/Containers/Data/Application/60F7254B-9367-4ADF-9A24-6D0261486EAD/Documents/Cache/httpbin.org/get/index.html_foo=bar

07/28/2016 22:54:43:207 Oauth2Test)[83082:.] EVURLCache.swift(88) cachedResponseForRequest:
	CACHE not found /Users/evermeer/Library/Developer/CoreSimulator/Devices/544A81ED-E3EB-40C9-B098-A695DF6B6A81/data/Containers/Bundle/Application/28AEFF4E-988B-45C5-A2B6-DA21DC53EB21/Oauth2Test.app/PreCache/httpbin.org/get/index.html_foo=bar

07/28/2016 22:55:03:397 Oauth2Test)[83082:.] EVURLCache.swift(165) storeCachedResponse(_:forRequest:):
	Writing data to /Users/evermeer/Library/Developer/CoreSimulator/Devices/544A81ED-E3EB-40C9-B098-A695DF6B6A81/data/Containers/Data/Application/60F7254B-9367-4ADF-9A24-6D0261486EAD/Documents/Cache/httpbin.org/get/index.html_foo=bar

07/28/2016 22:55:05:596 Oauth2Test)[83082:.] EVURLCache.swift(169) storeCachedResponse(_:forRequest:):
	CACHE save file to Cache  : /Users/evermeer/Library/Developer/CoreSimulator/Devices/544A81ED-E3EB-40C9-B098-A695DF6B6A81/data/Containers/Data/Application/60F7254B-9367-4ADF-9A24-6D0261486EAD/Documents/Cache/httpbin.org/get/index.html_foo=bar

(Optional(<NSMutableURLRequest: 0x7f9510649940> { URL: https://httpbin.org/get?foo=bar }), Optional(<NSHTTPURLResponse: 0x7f9511b00160> { URL: https://httpbin.org/get?foo=bar } { status code: 200, headers {
    "Access-Control-Allow-Origin" = "*";
    "Content-Length" = 395;
    "Content-Type" = "application/json";
    Date = "Thu, 28 Jul 2016 20:54:43 GMT";
    Server = nginx;
    "access-control-allow-credentials" = true;
} }), Optional(<7b0a2020 22617267 73223a20 7b0a2020 20202266 6f6f223a 20226261 72220a20 207d2c20 0a202022 68656164 65727322 3a207b0a 20202020 22416363 65707422 3a20222a 2f2a222c 200a2020 20202241 63636570 742d456e 636f6469 6e67223a 2022677a 69703b71 3d312e30 2c20636f 6d707265 73733b71 3d302e35 222c200a 20202020 22416363 6570742d 4c616e67 75616765 223a2022 6e6c2d42 453b713d 312e302c 206e6c2d 55533b71 3d302e39 2c206672 2d55533b 713d302e 382c2064 652d5553 3b713d30 2e372c20 656e2d55 533b713d 302e3622 2c200a20 20202022 486f7374 223a2022 68747470 62696e2e 6f726722 2c200a20 20202022 55736572 2d416765 6e74223a 20224f61 75746832 54657374 2f6e6c2e 65766963 742e4f61 75746832 54657374 2028313b 20694f53 20392e33 2e302922 0a20207d 2c200a20 20226f72 6967696e 223a2022 3231372e 3132332e 36392e31 3737222c 200a2020 2275726c 223a2022 68747470 733a2f2f 68747470 62696e2e 6f72672f 6765743f 666f6f3d 62617222 0a7d0a>), nil)
```
As you can see it will first look in the PreCache folder, then the Cashe folder, and then wirte it to the Cashe folder and then report that it's written and output the rusult in the Alamofire callback.

And the second time it's called this will be in the log:

```
07/28/2016 22:57:52:684 Oauth2Test)[83082:.] EVURLCache.swift(104) cachedResponseForRequest:
	Returning cached data from /Users/evermeer/Library/Developer/CoreSimulator/Devices/544A81ED-E3EB-40C9-B098-A695DF6B6A81/data/Containers/Data/Application/60F7254B-9367-4ADF-9A24-6D0261486EAD/Documents/Cache/httpbin.org/get/index.html_foo=bar

07/28/2016 22:58:05:165 Oauth2Test)[83082:.] EVURLCache.swift(159) storeCachedResponse(_:forRequest:):
	CACHE not rewriting stored file

(Optional(<NSMutableURLRequest: 0x7f9510613d40> { URL: https://httpbin.org/get?foo=bar }), nil, Optional(<7b0a2020 22617267 73223a20 7b0a2020 20202266 6f6f223a 20226261 72220a20 207d2c20 0a202022 68656164 65727322 3a207b0a 20202020 22416363 65707422 3a20222a 2f2a222c 200a2020 20202241 63636570 742d456e 636f6469 6e67223a 2022677a 69703b71 3d312e30 2c20636f 6d707265 73733b71 3d302e35 222c200a 20202020 22416363 6570742d 4c616e67 75616765 223a2022 6e6c2d42 453b713d 312e302c 206e6c2d 55533b71 3d302e39 2c206672 2d55533b 713d302e 382c2064 652d5553 3b713d30 2e372c20 656e2d55 533b713d 302e3622 2c200a20 20202022 486f7374 223a2022 68747470 62696e2e 6f726722 2c200a20 20202022 55736572 2d416765 6e74223a 20224f61 75746832 54657374 2f6e6c2e 65766963 742e4f61 75746832 54657374 2028313b 20694f53 20392e33 2e302922 0a20207d 2c200a20 20226f72 6967696e 223a2022 3231372e 3132332e 36392e31 3737222c 200a2020 2275726c 223a2022 68747470 733a2f2f 68747470 62696e2e 6f72672f 6765743f 666f6f3d 62617222 0a7d0a>), nil)
```
Where you can see it found the file in the Cache and it won't rewrite the same file, It will output the same response in the Alamofire callback.

**Comment by evermeer** [comment]
If you still have a problem with this, then just let me know...

**Comment by nmDejamobile** [comment]
Hi,

I can't say if it's fixed for the moment because of issue #29 that prevents me from using the cache.

Cache was created and worked before version 1.0, only callback were failing. Now cache is not even written.

Thanks.

**Comment by nmDejamobile** [comment]
As issue #29 is closed, this is all good for me.
